### PR TITLE
Add Debian 12 support

### PR DESCRIPTION
**Pull request by Justin-Fernbaugh**
Add support for Debian 12.

### CODE REVIEW

File: recipes/default.rb [hunk 1]
```diff
@@ -24,6 +24,8 @@
   ports %w(9323)
 end
 
+package %w(cron)
```
Comment (ramereth) [inline recipes/default.rb:27]: This unfortunately doesn't work with CentOS 7 as the package name doesn't. We already pull this in via ``base::default`` but I'd rather we didn't pull that completely in. Instead let's do the following:
1. Add a dep for the `cron` cookbook
2. Add the following resources:
```ruby
cron_package 'osl-docker'
cron_service 'osl-docker' do
  action [:enable, :start]
end
```

This should in theory work on all supported platforms and won't conflict with what we do in base.

Please make sure you update the tests as well.

**Review by ramereth — CHANGES_REQUESTED** [review]

File: Berksfile [hunk 2]
```diff
@@ -2,6 +2,7 @@ source 'https://supermarket.chef.io'
 
 solver :ruby, :required
 
+cookbook 'cron', git: 'git@github.com:sous-chefs/cron', branch: 'main'
```
Comment (ramereth) [inline Berksfile:5]: As I mentioned on the other PR, please remove this and just let Berkshelf pull this from the supermaket.

File: metadata.rb [hunk 3]
```diff
@@ -10,6 +10,7 @@
 
 depends          'apt'
 depends          'certificate'
+depends          'cron', '~> 7.0.20'
```
Comment (ramereth) [inline metadata.rb:13]: Please don't set the version as we'll run into issues with having to bump this on the base cookbook. I'd rather we just do it only at the base cookbook level.

**Review by ramereth — CHANGES_REQUESTED** [review]

File: metadata.rb (hunk 3 above)
Comment (Justin-Fernbaugh) [reply metadata.rb]: Updated the metadata to remove the version

File: Berksfile (hunk 2 above)
Comment (Justin-Fernbaugh) [reply Berksfile]: Removed cron from berks

File: metadata.rb [hunk 4]
```diff
@@ -10,7 +10,6 @@
 
 depends          'apt'
 depends          'certificate'
-depends          'cron', '~> 7.0.20'
```
Comment (ramereth) [inline metadata.rb]: You still need to keep this, just remove the version constraint please.

Comment (Justin-Fernbaugh) [reply metadata.rb]: Ah yes I added this back yesterday briefly after removing the whole line by accident.

File: spec/unit/recipes/default_spec.rb [hunk 5]
```diff
@@ -83,6 +83,8 @@
         )
       end
 
+      it { expect(chef_run).to install_cron_package('osl-docker') }
```
Comment (ramereth) [inline spec/unit/recipes/default_spec.rb:86]: Please also add the test for enabling/starting the service.

**Review by ramereth — CHANGES_REQUESTED** [review]

File: spec/unit/recipes/default_spec.rb (hunk 5 above)
Comment (Justin-Fernbaugh) [reply spec/unit/recipes/default_spec.rb:86]: Pushed a change to check if the service is enabled/started

**Review by ramereth — APPROVED** [review]
Minor bump

**Comment by Justin-Fernbaugh** [comment]
!bump minor ~

**Comment by osuosl-manatee** [comment]
Jenkins has merged this PR into `master` and has automatically performed a minor-level version bump to v4.11.0.  Have a nice day!

**Pull request merged**